inc/integrations/class-coral.php

```diff
@@ -378,25 +392,38 @@ private function get_username( $id ) : string {
 	 * @param string $id The SSO ID of the user.
 	 * @return int The post ID, or 0 if none is found.
 	 */
-	private function get_username_post_id( $id ) : int {
+	private function get_username_post_id( $id ): int {
 		global $wpdb;
 
+		$key            = "get_username_post_id_{$id}";
+		$stored_post_id = get_transient( $key );
+
+		if ( ! empty( $stored_post_id ) ) {
+			return $stored_post_id;
+		}
+
 		$post_id = $wpdb->get_var( // phpcs:ignore WordPress.DB.DirectDatabaseQuery
-			$wpdb->prepare( 
-				"SELECT ID 
-					FROM {$wpdb->posts} 
-				WHERE 
+			$wpdb->prepare(
+				"SELECT ID
+					FROM {$wpdb->posts}
+				WHERE
 					post_title=%s AND
 					post_type=%s AND
 					post_status='publish'
-				LIMIT 1 ",
+				LIMIT 1",
 				[
 					$id,
 					$this->post_type,
 				]
 			)
 		);
 
+		// Store the post ID in a transient.
+		// This can be set forever since the SSO ID to post ID relationship will never change.
```

Comment:
```suggestion
		/*
		 * Store the post ID in a transient.
		 * This can be set forever since the SSO ID to post ID relationship will never change.
		 */
```